### PR TITLE
[SM-711] API Update command addition

### PR DESCRIPTION
**Pull request by NachoRaik**
Addition of pages referring to updating APIs through ASM.

### CODE REVIEW

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+=== Update API instance in API Manager
+To update an existing API instance, perform the following steps:
+
+. Go to *Anypoint Platform > API Manager*
+. Search and select your API instance in the table. 
++
+Once inside the API details page, you can perform the following actions:
+. Update the instance tags, at the *API Configuration* section.
+. Update the asset version it points to. Note that you can only update if there is a newer version of the asset, that belongs to the same product version.
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Sure? I think you could decrease change the version to a lower one. I think I would remove the `Note...` part.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+--specPath=_(Optional) path to the API specification file_ \
+--specTags=_list of comma separated tags to be added to the asset on Exchange_ \
+--instanceTags=_list of comma separated tags to be added to the instance on API Manager_ 
+----
++
+[%header%autowidth.spread]
+|===
+| Parameter Name | Description | Required or Optional | Default Value
+| `name` | The name of the Kubernetes API resource that you want to provision, for example, `orders-api`. | Required | Not applicable
+| `namespace` | The namespace in which you want to create the adapter, for example, `mulesoft-apps`. | Optional | default
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Can you check if the namespace remains optional? I remember Rodri had removed the optional part for many of the commands.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+| `environmentId` | The ID of the environment where you want to create your API instance. | Required | Not applicable
+| `groupId` | The ID of the group of assets on Exchange, for example, the organization ID. You can find the organization ID on the API Manager landing page or in Access Management. | Optional | Organization ID of the organization that contains the environment.
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
```
Remove group ID

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+----
++
+[%header%autowidth.spread]
+|===
+| Parameter Name | Description | Required or Optional | Default Value
+| `name` | The name of the Kubernetes API resource that you want to provision, for example, `orders-api`. | Required | Not applicable
+| `namespace` | The namespace in which you want to create the adapter, for example, `mulesoft-apps`. | Optional | default
+| `environmentId` | The ID of the environment where you want to create your API instance. | Required | Not applicable
+| `groupId` | The ID of the group of assets on Exchange, for example, the organization ID. You can find the organization ID on the API Manager landing page or in Access Management. | Optional | Organization ID of the organization that contains the environment. 
+| `assetId` | The ID of the assets that are a part of an organization. Supports alphanumeric characters without spaces. | Required | Not applicable
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
```
Remove asset ID

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+| `version` | The version of the asset on Exchange. The version number must follow the semantic versioning format for major, minor, and patch releases. | Required | Not applicable
+| `instanceLabel` | The ID of the API instances of an Asset within a given environment. Supports alphanumeric characters without spaces. | Required | Not applicable
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
```
Remove instance label

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+| `specTags` | Tags to add to the asset on Exchange. |  Optional | Not applicable
+| `instanceTags` | Tags to add to the API instance on API Manager. |  Optional | Not applicable
+| `user` | The username credential of the Anypoint Platform user who is creating the API. | Required | Not applicable
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
```
Remove user

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+| `password` | The password of the Anypoint Platform user who is creating the API. | Required | Not applicable
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
```
Remove password

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+|===
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: You need to add the new fields to the table, right?

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
++
+. Verify the API status using the following command:
+`asmctl api list`
++
+The API that was created is displayed.
+. Optionally, you can verify that an asset is created on Exchange and an API is created on API Manager.
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Rewrite ☝️

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+[[create-api-ASM-CLI-ConnAps]]
+=== Autocreate by Using Your Connected Apps Credentials
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Why this is included on this PR?

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+. Search and select your API instance in the table. 
++
+Once inside the API details page, you can perform the following actions:
+. Update the instance tags, at the *API Configuration* section.
+. Update the asset version it points to. Note that you can only update if there is a newer version of the asset, that belongs to the same product version.
+image::update-api-asset.png[Display asset update]
+
+
+[[create-api-CRD]]
+== Create an API using the Anypoint Service Mesh
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc]: Create or update?

**Review by tbruno — COMMENTED** [review]
Thanks for starting this. Some questions Nacho

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 1 above)
Comment (NachoRaik) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: This actually has a weird behavior and was doubting of adding it.
We can move to a lower version, _but_ the button is not enabled if you are on the latest one (in that case, it just shows the "Latest" version).

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 2 above)
Comment (NachoRaik) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Yes, this is what we were discussing the other day. This whole table must be deleted though, the correct table is at the bottom of the page.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 1 above)
Comment (tbruno) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: ahh, I understand. Still, I would remove the note part. Agree that the behaviour is not totally clear.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 12]
```diff
@@ -0,0 +1,86 @@
+= Update an API Using the CLI
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: service mesh, microservices, updating, installing
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
:keywords: service mesh, microservices, updating
```

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+APIs in the Platform can be updated -both assets and API instances-, from creating a whole new version (maintaining the product version) or just their tags. You can update an API using Anypoint Platform or Anypoint Service Mesh:
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
APIs are constantly evolving. Anypoint Service Mesh allows you to update the specification of your API. You can update an API using Anypoint Platform or Anypoint Service Mesh:
```
IMHO, I wouldn't give much importance to the tags update.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc [hunk 14]
```diff
@@ -0,0 +1,66 @@
+= Update an API Using a CRD
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: service mesh, microservices, updating, installing
+
+APIs in the Platform can be updated -both assets and API instances-, from creating a whole new version (maintaining the product version) or just their tags. You can update an API using Anypoint Platform or Anypoint Service Mesh:
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc]:
```suggestion
APIs are constantly evolving. Anypoint Service Mesh allows you to update the specification of your API. You can update an API using Anypoint Platform or Anypoint Service Mesh:
```
IMHO, I wouldn't give much importance to the tags update.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc [hunk 15]
```diff
@@ -0,0 +1,66 @@
+= Update an API Using a CRD
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: service mesh, microservices, updating, installing
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc]:
```suggestion
:keywords: service mesh, microservices, updating
```

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+
+* <<update-api-AP, Update an API using Anypoint Platform>>
++
+New versions of an API asset can be created via Exchange. Then, API instances can be updated to use this new version of the asset.
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]:
```suggestion
Once a new version of an API is published to Exchange, the API instances associated with the asset can be updated to use this new version of the asset.
```

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+
+APIs in the Platform can be updated -both assets and API instances-, from creating a whole new version (maintaining the product version) or just their tags. You can update an API using Anypoint Platform or Anypoint Service Mesh:
+
+* <<update-api-AP, Update an API using Anypoint Platform>>
++
+New versions of an API asset can be created via Exchange. Then, API instances can be updated to use this new version of the asset.
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CRD.adoc]:
```suggestion
Once a new version of an API is published to Exchange, the API instances associated with the asset can be updated to use this new version of the asset.
```

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+[source,text,linenums]
+----
+asmctl api update  
+--name=_Kubernetes API resource name_ \
+--namespace=_namespace name_ \
+--version=_(Optional) new asset version for the API_ \
+--mainFile=_(Optional) name of the main file in the API specification_ \
+--specPath=_(Optional) path to the API specification file_ \
+--specTags=_list of comma separated tags to be added to the asset on Exchange_ \
+--instanceTags=_list of comma separated tags to be added to the instance on API Manager_
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc]: Aren't instance tags and spec tags optional? Should we put `(Optional)` like with the rest?

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+| `instanceTags` | The tags to add to the API instance on API Manager |  Optional | Not applicable
+|===
++
+. Verify the API status:
+`asmctl api list`
++
+The new API is shown as `Ready`.
+. Optionally, verify the changes in the Exchange asset and in the API instance in API Manager.
+
+== See Also
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc:83]: Should we put that you can only bump minor, patch versions?

**Review by tbruno — COMMENTED** [review]
Good progress!

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 19 above)
Comment (NachoRaik) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc:83]: I thought I had add it somewhere 😅 .
Now it is included in the sentence at the beginning of the page.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+. Go to *Anypoint Platform > API Manager*
+. Search and select your API instance in the table. 
++
+Once inside the API details page, you can perform the following actions:
+. Update the instance tags, at the *API Configuration* section.
+. Update the asset version it points to.
+image::update-api-asset.png[Display asset update]
+
+
+[[update-api-cli]]
```
Comment (tbruno) [inline modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc:19]: One last, comment,  I would reorder and put the update of service mesh before the updates on the platform.

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 19 above)
Comment (tbruno) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc:83]: Thanks :)

**Review by tbruno — APPROVED** [review]
LGTM 👍 
Thanks for applying the feedback

File: modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc (hunk 20 above)
Comment (priyaayyar) [reply modules/ROOT/pages/update-an-api-configure-service-mesh-CLI.adoc:19]: Done! Thanks!

**Pull request merged**